### PR TITLE
Feature/cont ecommerce

### PR DESCRIPTION
**Pull request by JoanSerna**

### CODE REVIEW

File: apps/guest/src/app/app.module.ts [hunk 1]
```diff
@@ -44,7 +44,8 @@ import { UserEffects } from 'guest/app/reducers/user/user.effects';
 import { registerLocaleData } from '@angular/common';
 const app = firebase.initializeApp(environment.fire, 'app');
 if (environment.emulate) {
-  app.auth().useEmulator('http://localhost:9099/');
+  // @ts-ignore
```
Comment (ruedagato) [inline apps/guest/src/app/app.module.ts:47]: why this is necessary?

File: apps/guest/src/app/app.module.ts [hunk 2]
```diff
@@ -44,7 +44,8 @@ import { UserEffects } from 'guest/app/reducers/user/user.effects';
...
+  app.auth().useEmulator('http://localhost:9099/', { disableWarnings: true });
```
Comment (ruedagato) [inline apps/guest/src/app/app.module.ts:48]: keep this change locally

File: apps/guest/src/ecommerce/ecommerce.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (ruedagato) [inline apps/guest/src/ecommerce/ecommerce.component.spec.ts:1]: delete this file if you are not use unit test

File: apps/guest/src/ecommerce/ecommerce.component.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      .pipe(selectUserState)
+      .pipe(first())
+      .subscribe(({ user, hotel }) => {
+        const {
+          order: { value },
+        } = this.orderForm.controls;
+        const orders = value.filter((o) => o.quantity);
+        ecommerce.categories = null;
+        ecommerce.hotel = null;
+        console.log({ guest: user, hotel, orders, ecommerce });
```
Comment (ruedagato) [inline apps/guest/src/ecommerce/ecommerce.component.ts:73]: remove log

File: apps/guest/src/home/pages/welcome/welcome.component.ts [hunk 5]
```diff
@@ -14,7 +14,11 @@ import { RequestService } from '@contler/dynamic-services';
 import { Store } from '@ngrx/store';
 import { State } from 'guest/app/reducers';
 import { selectUserState } from 'guest/app/reducers/user/user.selectors';
+import { EcommerceEntity } from '@contler/entity/ecommerce.entity';
 
+class EcommerceOption extends EcommerceEntity {
```
Comment (ruedagato) [inline apps/guest/src/home/pages/welcome/welcome.component.ts:19]: move it to a external file following architecture structure project

File: apps/guest/src/home/pages/welcome/welcome.component.ts [hunk 6]
```diff
@@ -41,6 +46,22 @@ export class WelcomeComponent implements OnInit {
       .pipe(selectUserState)
       .pipe(map((data) => data.hotel))
       .pipe(switchMap((hotel) => this.specialZoneGuestService.getSpecialZoneGuestActive(hotel.uid)));
+
+    this.ecommerce$ = this.store
```
Comment (ruedagato) [inline apps/guest/src/home/pages/welcome/welcome.component.ts:50]: change the logic to get all e-commerce from a hotel for a different service due to get all e-commerce in the first charge with the user info can generate bad performance of the first charge

Comment (ruedagato) [reply apps/guest/src/home/pages/welcome/welcome.component.ts:50]: you can create a new  redux with an effect that detects when user is login and user the hotel data to send a request to the backend for get e-commerce info

File: apps/hotel/src/ecommerce/create-ecommerce/create-ecommerce.component.html [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+        <h3 class="title">{{ 'ecommerce.products.text' | translate }}</h3>
+        <div class="section" [formGroup]="formProduct">
+          <mat-form-field appearance="fill">
+            <input type="text" matInput formControlName="name" />
+            <mat-hint>{{ 'ecommerce.products.nameHint' | translate }}</mat-hint>
+          </mat-form-field>
+          <mat-form-field appearance="fill">
+            <mat-select formControlName="category">
+              <ng-container *ngFor="let categoryControl of categories.controls">
+                <mat-option *ngIf="categoryControl.valid" [value]="categoryControl.get('name').value">{{
```
Comment (ruedagato) [inline apps/hotel/src/ecommerce/create-ecommerce/create-ecommerce.component.html:63]: fix identation

File: apps/hotel/src/ecommerce/create-ecommerce/create-ecommerce.component.scss [hunk 8]
```diff
@@ -0,0 +1,170 @@
+.two_columns {
+  display: grid;
+  grid-template-columns: 70% 30%;
+  grid-column-gap: 50px;
+  margin-bottom: 40px;
+}
+
+.title {
+  font-size: 15.2px;
```
Comment (ruedagato) [inline apps/hotel/src/ecommerce/create-ecommerce/create-ecommerce.component.scss:9]: use global mixin text https://github.com/Contler/monorepo-web/blob/fa48d59cfe05f1417c2345c4d4d028963403d5f5/libs/styles/_mixin.scss#L1

File: apps/hotel/src/ecommerce/create-ecommerce/create-ecommerce.component.scss [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+  display: grid;
+  grid-template-columns: 1fr 1fr 1fr;
+  align-items: center;
+}
+
+.text-left {
+  text-align: left;
+}
+// Overwrite style angular material on expansion panel tag
+.accordion {
```
Comment (ruedagato) [inline apps/hotel/src/ecommerce/create-ecommerce/create-ecommerce.component.scss:168]: move it class to https://github.com/Contler/monorepo-web/blob/b3adbe8bca20485558edbf38073b7c18e8b88136/libs/styles/_material.scss and the end an check if with it you can delete !import

File: apps/hotel/src/ecommerce/create-ecommerce/create-ecommerce.component.spec.ts [hunk 10]
```diff
@@ -0,0 +1,24 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (ruedagato) [inline apps/hotel/src/ecommerce/create-ecommerce/create-ecommerce.component.spec.ts:1]: remove it if it you are not use unit test

File: apps/hotel/src/preferences/pages/guest-home/guest-home.component.ts [hunk 11]
```diff
@@ -36,6 +38,8 @@ export class GuestHomeComponent implements OnInit {
     private router: Router,
     private matDialog: MatDialog,
     public translate: TranslateService,
+    private ecommerceService: EcommerceService,
+    private changeDetectorRef: ChangeDetectorRef,
```
Comment (ruedagato) [inline apps/hotel/src/preferences/pages/guest-home/guest-home.component.ts:42]: it is not in use

File: apps/hotel/src/preferences/pages/guest-home/guest-home.component.ts [hunk 12]
```diff
@@ -1,4 +1,4 @@
-import { Component, HostListener, OnInit } from '@angular/core';
+import { ChangeDetectorRef, Component, HostListener, OnInit } from '@angular/core';
```
Comment (ruedagato) [inline apps/hotel/src/preferences/pages/guest-home/guest-home.component.ts:1]: ChangeDetectorRef is not in use

File: apps/guest/src/app/app.module.ts (hunk 1 above)
Comment (JoanSerna) [reply apps/guest/src/app/app.module.ts:47]: For remove alert emulator on guest app

File: apps/guest/src/app/app.module.ts (hunk 2 above)
Comment (JoanSerna) [reply apps/guest/src/app/app.module.ts:48]: environment.emulate is only true when app is running locally ?

**Pull request closed without merging**